src/store/epics/session.ts

```diff
@@ -0,0 +1,18 @@
+import { GoogleSignin } from '@react-native-google-signin/google-signin';
+import { combineEpics, ofType } from 'redux-observable';
+import { pluck, switchMap } from 'rxjs';
+import { LOGIN, SIGN_IN_GOOGLE, successAction } from '../actionTypes';
+import { EpicType } from './types';
+
+export const watchGoogleSignIn: EpicType = action$ =>
+  action$.pipe(
+    ofType(successAction(SIGN_IN_GOOGLE)),
+    pluck('payload'),
+    switchMap(async (data: any) => {
+      const { user } = data;
+      const { accessToken, idToken } = await GoogleSignin.getTokens();
```

Comment:
Minor: I would say that you can pass only this method to `switchMap`, then use `map` to return new action